content/docs/reference/pkg/aws/s3/bucket.md

```diff
@@ -0,0 +1,10931 @@
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Policy</td>
+            <td class="align-top">
+                
+                <code>string?</code>
+            </td>
+            <td class="align-top">{{% md %}} A valid [bucket policy](https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html) JSON document.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Region</td>
+            <td class="align-top">
+                
+                <code>string</code>
+            </td>
+            <td class="align-top">{{% md %}} If specified, the AWS region this bucket should reside in. Otherwise, the region used by the callee.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Replication<wbr>Configuration</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketreplicationconfiguration">Pulumi.<wbr>Aws.<wbr>S3.<wbr>Bucket<wbr>Replication<wbr>Configuration?</a></code>
+            </td>
+            <td class="align-top">{{% md %}} A configuration of [replication configuration](http://docs.aws.amazon.com/AmazonS3/latest/dev/crr.html) (documented below).
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Request<wbr>Payer</td>
+            <td class="align-top">
+                
+                <code>string</code>
+            </td>
+            <td class="align-top">{{% md %}} Specifies who should bear the cost of Amazon S3 data transfer.
+Can be either `BucketOwner` or `Requester`. By default, the owner of the S3 bucket would incur
+the costs of any data transfer. See [Requester Pays Buckets](http://docs.aws.amazon.com/AmazonS3/latest/dev/RequesterPaysBuckets.html)
+developer guide for more information.
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Server<wbr>Side<wbr>Encryption<wbr>Configuration</td>
+            <td class="align-top">
+                
+                <code><a href="#bucketserversideencryptionconfiguration">Pulumi.<wbr>Aws.<wbr>S3.<wbr>Bucket<wbr>Server<wbr>Side<wbr>Encryption<wbr>Configuration?</a></code>
+            </td>
+            <td class="align-top">{{% md %}} A configuration of [server-side encryption configuration](http://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html) (documented below)
+ {{% /md %}}
+
+            
+            </td>
+        </tr>
+    
+        <tr>
+            <td class="align-top">Tags</td>
+            <td class="align-top">
+                
+                <code>Dictionary&lt;string, object&gt;?</code>
+            </td>
+            <td class="align-top">{{% md %}} A mapping of tags that identifies subset of objects to which the rule applies.
+The rule applies only to objects having all t
```

Comment:
Reminder: Pretty sure for these nested types, we'll need to change these from snake_case to camelCase. @pgavlin to confirm that's still the case.

Reply:
I created https://github.com/pulumi/pulumi/issues/4095 to track this. I will confirm with Pat offline and fix it in a separate PR in the pulumi/pulumi repo.